app/src/main/res/layout/eth_tx_detail.xml

```diff
@@ -276,37 +286,112 @@
             </LinearLayout>
 
             <LinearLayout
-                android:id="@+id/data"
+                android:id="@+id/undecode_data"
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
-                android:orientation="vertical">
+                android:orientation="vertical"
+                android:visibility="gone">
 
                 <TextView
                     android:layout_width="wrap_content"
                     android:layout_height="wrap_content"
                     android:layout_marginStart="16dp"
                     android:layout_marginTop="14dp"
-                    android:text="Data"
+                    android:text="Input_Data"
                     android:textColor="@color/white40"
                     android:textSize="14sp"
                     android:textStyle="bold" />
 
+                <TextView
+                    android:id="@+id/input_data"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_marginHorizontal="16dp"
+                    android:breakStrategy="simple"
+                    android:textColor="@color/white"
+                    android:textStyle="bold"
+                    tools:text="0x35623ea315322ff2618e7469c2e88415111b5444000000000000000000000asd" />
+
+                <LinearLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:orientation="horizontal"
+                    android:gravity="center_vertical">
+
+                    <ImageView
+                        android:layout_width="15dp"
+                        android:layout_height="15dp"
+                        android:layout_marginStart="16dp"
+                        android:src="@drawable/direction" />
+
+                    <TextView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_marginStart="3dp"
+                        android:layout_marginTop="3dp"
+                        android:text="@string/undecod_info"
+                        android:textColor="#FF0B0B"
```

Comment:
add new color in https://github.com/KeystoneHQ/Keystone-cold-app/blob/main/app/src/main/res/values/colors.xml